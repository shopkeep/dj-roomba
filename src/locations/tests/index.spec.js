const locations = require("../");
const mockLocations = require("./test-data/locations");
let mockTimezone;

jest.mock("@shopkeep/bot-scripts/utils", () => {
  return {
    getUser: jest.fn(() => ({ id: "test-user", tz: mockTimezone }))
  };
});

jest.mock("../utils", () => {
  return {
    getLocations: jest.fn(() => mockLocations),
    createPlayer: jest.fn(() => "mock-player")
  };
});

describe("locations", function() {
  let testContext;

  beforeEach(function() {
    mockTimezone = undefined;
    testContext = {};
    testContext.locations = locations();
    testContext.botMock = { botMock: true };
    testContext.messageMock = { user: "test-user" };
  });

  describe("getLocations", function() {
    it("returns a list of locations", function() {
      expect(testContext.locations.getLocations()).toMatchSnapshot();
    });
  });

  describe("getLocation", function() {
    describe("when matching on location name", function() {
      beforeEach(function() {
        mockTimezone = undefined;
        testContext.locationName = "New York";
      });

      it("returns a matching location", async function() {
        expect(
          await testContext.locations.getLocation(
            testContext.botMock,
            testContext.messageMock,
            testContext.locationName
          )
        ).toMatchSnapshot();
      });
    });

    describe("when matching on users timezone", function() {
      beforeEach(function() {
        mockTimezone = "Europe/London";
        testContext.locationName = undefined;
      });

      it("returns a matching location", async function() {
        expect(
          await testContext.locations.getLocation(
            testContext.botMock,
            testContext.messageMock,
            testContext.locationName
          )
        ).toMatchSnapshot();
      });
    });

    describe("when no location match", function() {
      beforeEach(function() {
        mockTimezone = "Asia/Shanghai";
        testContext.locationName = "Shanghai";
      });

      it("returns a matching location", async function() {
        expect(
          await testContext.locations.getLocation(
            testContext.botMock,
            testContext.messageMock,
            testContext.locationName
          )
        ).toEqual(null);
      });
    });
  });
});
