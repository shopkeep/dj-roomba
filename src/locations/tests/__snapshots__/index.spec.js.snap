// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`locations getLocation when matching on location name returns a matching location 1`] = `
Object {
  "name": "New York",
  "player": "mock-player",
  "spotifyMarket": "US",
}
`;

exports[`locations getLocation when matching on users timezone returns a matching location 1`] = `
Object {
  "name": "London",
  "player": "mock-player",
  "spotifyMarket": "GB",
}
`;

exports[`locations getLocations returns a list of locations 1`] = `
Array [
  Object {
    "defaultVolume": "42",
    "name": "New York",
    "sonosAddress": "127.0.0.1",
    "spotifyMarket": "US",
    "spotifyRegion": "US",
    "timezone": "America/New_York",
  },
  Object {
    "defaultVolume": "50",
    "name": "London",
    "sonosAddress": "127.0.0.1",
    "spotifyMarket": "GB",
    "spotifyRegion": "EU",
    "timezone": "Europe/London",
  },
]
`;
